Sources/ArcGISToolkit/FloorFilter/FloorFilterViewModel.swift

```diff
@@ -0,0 +1,118 @@
+internal class FloorFilterViewModel {
+    
+    // The MapView, Map and Floor Manager are set in the FloorFilterViewController when the map is loaded
+    public var mapView: AGSMapView?
+    public var map: AGSMap?
+    public var floorManager: AGSFloorManager?
+    
+    public var sites: [AGSFloorSite] {
+        return floorManager?.sites ?? []
+    }
+    
+    // Facilities in the site that is selected
+    // If no site is selected then the list is empty
+    // If the sites data does not exist in the map, then use all the facilities in the map
+    public var facilities: [AGSFloorFacility] {
+        guard let floorManager = floorManager else { return [] }
+        return sites.isEmpty ? floorManager.facilities : floorManager.facilities.filter { $0.site == selectedSite }
+    }
+    
+    // Levels that are visible in the expanded Floor Filter levels table view
+    // Reverse the order of the levels to make it in ascending order
+    public var visibleLevelsInExpandedList: [AGSFloorLevel] {
+        guard let floorManager = floorManager else { return [] }
+        return facilities.isEmpty ? floorManager.levels : floorManager.levels.filter { $0.facility == selectedFacility }.reversed()
+    }
+    
+    // All the levels in the map
+    public var allLevels: [AGSFloorLevel] {
+        return floorManager?.levels ?? []
+    }
+    
+    // The site, facility, and level that are selected by the user
+    public var selectedSite: AGSFloorSite?
+    public var selectedFacility: AGSFloorFacility?
+    public var selectedLevel: AGSFloorLevel?
+    
+    // The default vertical order is 0 according to Runtime 100.12 update for AGSFloorManager
+    public let defaultVerticalOrder = 0
+
+    public func reset() {
+        floorManager = nil
+        selectedSite = nil
+        selectedFacility = nil
+        selectedLevel = nil
+    }
+    
+    public func getSelectedSite() -> AGSFloorSite? {
+        return sites.first { $0 == selectedSite }
+    }
+
+    public func getSelectedFacility() -> AGSFloorFacility? {
+        return facilities.first { $0 == selectedFacility }
+    }
+    
+    public func getSelectedVisibleLevel() -> AGSFloorLevel? {
+        return visibleLevelsInExpandedList.first { $0 == selectedLevel }
+    }
+    
+    // Sets the visibility of all the levels on the map based on the vertical order of the current selected level
+    public func filterMapToSelectedLevel() {
+        guard let selectedLevel = getSelectedVisibleLevel() ?? selectedLevel else { return }
+        allLevels.forEach {
+            $0.isVisible = $0.verticalOrder == selectedLevel.verticalOrder
+        }
+    }
+    
+    // Zooms to the facility if there is a selected facility, otherwise zooms to the site if there is no selected facility
+    public func zoomToSelection() {
+        if let _ = selectedFacility {
+            zoomToFacility()
+        } else {
+            if let _ = selectedSite {
+                zoomToSite()
+            }
+        }
```

Comment:
It's generally preferred to check agains `nil` as opposed to an empty `let`.
```suggestion
        if selectedFacility != nil {
            zoomToFacility()
        } else if selectedSite != nil {
            zoomToSite()
        }
```